apps/srt-file-transmit.cpp

```diff
@@ -59,110 +51,172 @@ void OnINT_ForceExit(int)
+        { o_version,      OptionScheme::ARG_NONE }
+    };
 
-    string sf = Option<OutString>(params, "no", o_noflush);
-    if (sf == "" || !false_names.count(sf))
-        ::g_skip_flushing = true;
+    options_t params = ProcessOptions(argv, argc, optargs);
 
-    string sfull = Option<OutString>(params, "no", o_fullstats);
-    if (sfull == "" || !false_names.count(sfull))
-        ::transmit_total_stats = true;
+          bool print_help    = Option<OutBool>(params, false, o_help);
+    const bool print_version = Option<OutBool>(params, false, o_version);
 
-    string source = args[0];
-    string target = args[1];
+    if (params[""].size() != 2 && !print_help && !print_version)
+    {
+        cerr << "ERROR. Invalid syntax. Specify source and target URIs.\n";
+        if (params[""].size() > 0)
+        {
+            cerr << "The following options are passed without a key: ";
+            copy(params[""].begin(), params[""].end(), ostream_iterator<string>(cerr, ", "));
+            cerr << endl;
+        }
+        print_help = true; // Enable help to print it further
+    }
 
-    UriParser us(source), ut(target);
 
-    Verb() << "SOURCE type=" << us.scheme() << ", TARGET type=" << ut.scheme();
+    if (print_help)
+    {
+        cout << "SRT sample application to transmit files.\n";
+        cerr << "SRT Library version: " << SRT_VERSION << endl;
+        cerr << "Usage: srt-file-transmit [options] <input-uri> <output-uri>\n";
+        cerr << "\n";
+
+        PrintOptionHelp(o_chunk, "<chunk=1456>", "max size of data read in one step");
+        PrintOptionHelp(o_no_flush, "", "skip output file flushing");
+        PrintOptionHelp(o_bwreport, "<every_n_packets=0>", "bandwidth report frequency");
+        PrintOptionHelp(o_statsrep, "<every_n_packets=0>", "frequency of status report");
+        PrintOptionHelp(o_statsout, "<stats_output=stdout>", "stats output [stdout|filename]");
+        PrintOptionHelp(o_statspf, "<format=default>", "stats printing format [json|csv|default]");
+        PrintOptionHelp(o_statsfull, "", "full counters in stats-report (prints total statistics)");
+        PrintOptionHelp(o_loglevel, "<level=error>", "log level [fatal,error,info,note,warning]");
+        PrintOptionHelp(o_logfa, "<fas=general,...>", "log functional area [all,general,bstats,control,data,tsbpd,rexmit]");
+        PrintOptionHelp(o_logfile, "<filename="">", "write logs to file");
+        PrintOptionHelp(o_quiet, "", "quiet mode (default off)");
+        PrintOptionHelp(o_verbose, "", "verbose mode (default off)");
+        cerr << "\n";
+        cerr << "\t-h,-help - show this help\n";
+        cerr << "\t-version - print SRT library version\n";
+        cerr << "\n";
+        cerr << "\t<input-uri>  - [srt://<ip>:<port>?<query>,udp://<ip>:<port>?<query>,file:///<path>,file://con]\n";
+        cerr << "\t<output-uri> - [srt://<ip>:<port>?<query>,udp://<ip>:<port>?<query>,file:///<path>,file://con]\n";
```

Comment:
Very long lines. I think you better explain these arguments commonly as URI, giving syntax requirements and schemes supported.

Reply:
Changed to:
```
cerr << "\t<input-uri>  - [scheme://HOST:PORT/PATH?PARAM1=VALUE&PARAM2=VALUE]\n";
cerr << "\t<output-uri> - [scheme://HOST:PORT/PATH?PARAM1=VALUE&PARAM2=VALUE]\n";
cerr << "\tSupported schemes: srt, udp, file://con for stdout\n";
```

Reply:
Better, but I see no point in repeating the scheme twice. Simply announce that they both use the URI syntax, which would be explained just once.

Reply:
Accepted.